lib/generation/generators/pb_generator.dart

```diff
@@ -47,4 +48,23 @@ abstract class PBGenerator {
       )
     ''';
   }
+
+  void addTagToAnalytics(String tagName) {
+    if (MainInfo()
+        .amplitudMap['eventProperties']['tags']
+        .containsKey(tagName)) {
+      MainInfo().amplitudMap['eventProperties']['tags'][tagName]++;
+    } else {
+      MainInfo().amplitudMap['eventProperties']['tags'][tagName] = 1;
+    }
+    addToAnalytics('Number of custom generated');
+  }
+
+  void addToAnalytics(String propertyName) {
```

Comment:
This code has also been copy-pasted in a different file above. We should contain this logic to avoid copy-pasting